### PR TITLE
Upgrade to std::thread and std::chrono

### PR DESCRIPTION
**Pull request by Greg-Griffith**
The goal of this PR is to make some of the larger changes required to reduce our dependency on boost. IMO moving away from boost where possible helps to reduce dependency maintenance and the difficulties that can come with it. 

- changes our threading to std::thread and implements a thread_group for std::threads to replace boost::thread_group since there isnt one for std threads in the stl library. 
- changes our use of boost::chrono to std::chrono
- removes the using namespace std; from util.cpp
- 1 misc change moving from boost to std
- - inside "remove using namespace std;" two boost::call_once are changed to std::call_once

### CODE REVIEW

File: src/threadgroup.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#ifndef GRIFFITH_RED_THREAD_GROUP_H
```
Comment (sickpig) [inline src/threadgroup.h]: weird name, no? :smile_cat:

**Review by sickpig — COMMENTED** [review]
just skimmed quickly leave a nit comment. I wonder if we could split the PR in 3 parts, one related to the thread changes, one for chrono and on for the rest (I'm also ok leaving it as is thou)

File: src/threadgroup.h (hunk 1 above)
Comment (Greg-Griffith) [reply src/threadgroup.h]: woops. i imported that header from another program i wrote. forgot to change the define. will fix

**Comment by Greg-Griffith** [comment]
@sickpig it would be hard to do that and also have tests pass. thread sleeping requires both libraries. you cant use boost::chrono to set a time duration for a sleep in a std::thread and vice versa with std::chrono in boost::thread

**Comment by Greg-Griffith** [comment]
The removal of CScheduler in this PR also addresses @awemany's concerns in #1164

**Comment by sickpig** [comment]
It seems like we have a linker error during final step of win32 build process, see: 

https://travis-ci.org/BitcoinUnlimited/BitcoinUnlimited/jobs/478801616#L1659

**Comment by sickpig** [comment]
> @sickpig it would be hard to do that and also have tests pass. thread sleeping requires both libraries. you cant use boost::chrono to set a time duration for a sleep in a std::thread and vice versa with std::chrono in boost::thread

fair enough.

**Comment by ptschip** [comment]
64 bit compiles on native windows...

but no 32 bit:

.libs/libbitcoincash_la-utiltime.o: In function `sleep_for<long long int, std::ratio<1ll, 1000ll> >':
c:/MinGW/mingw32/i686-w64-mingw32/include/c++/thread:279: undefined reference to `_imp__nanosleep'

**Comment by Greg-Griffith** [comment]
@ptschip thanks for testing that

**Comment by ptschip** [comment]
you can try this...it works for win32:

#ifdef WIN32
#include <windows.h> // for Sleep()
#endif


void MilliSleep(int64_t n)
{
#ifdef WIN32
   Sleep(n);
#else
   std::this_thread::sleep_for(std::chrono::milliseconds(n));
#endif
}

**Comment by gandrewstone** [comment]
issues with our recent CB merge and this PR...

'blockrelay/compactblock.cpp' || echo '../../src/'`blockrelay/compactblock.cpp
../../src/blockrelay/compactblock.cpp:898:30: error: no member named 'setprecision' in namespace 'std'
    ss << std::fixed << std::setprecision(1);
                        ~~~~~^
../../src/blockrelay/compactblock.cpp:915:30: error: no member named 'setprecision' in namespace 'std'
    ss << std::fixed << std::setprecision(1);
                        ~~~~~^
../../src/blockrelay/compactblock.cpp:952:30: error: no member named 'setprecision' in namespace 'std'
    ss << std::fixed << std::setprecision(2);
                        ~~~~~^
../../src/blockrelay/compactblock.cpp:988:30: error: no member named 'setprecision' in namespace 'std'
    ss << std::fixed << std::setprecision(2);
                        ~~~~~^
../../src/blockrelay/compactblock.cpp:1004:30: error: no member named 'setprecision' in namespace 'std'
    ss << std::fixed << std::setprecision(1);
                        ~~~~~^
5 errors generated.

**Comment by Greg-Griffith** [comment]
this is a simple issue of a missing inclusion. will fix in a minute

File: src/net.cpp [hunk 2]
```diff
@@ -2467,28 +2528,28 @@ void StartNode(boost::thread_group &threadGroup, CScheduler &scheduler)
     // Start threads
     //
 
-    threadGroup.create_thread(boost::bind(&TraceThread<void (*)()>, "dnsseed", &ThreadAddressSeeding));
```
Comment (gandrewstone) [inline src/net.cpp:2470]: The thread name functionality is useful for debugging.  We can preserve it by calling "RenameThread" defined in util.h/cpp.  But this function must be called from within the running thread.

Although I'm only marking this one, this should be fixed for all the spawned threads.

**Pull request merged**